indigo_app/views/tasks.py

```diff
@@ -255,3 +255,38 @@ def post(self, request, *args, **kwargs):
         task.save()
 
         return redirect('task_detail', place=self.kwargs['place'], pk=self.kwargs['pk'])
+
+
+class TaskAssignView(TaskViewBase, View, SingleObjectMixin):
+    # permissions
+    permission_required = ('indigo_api.change_task',)
+
+    unassign = False
+    http_method_names = [u'post']
+    model = Task
+
+    def post(self, request, *args, **kwargs):
+        task = self.get_object()
+        user = self.request.user
+        task.updated_by_user = user
+        if self.unassign:
+            task.assigned_to = None
+            action.send(user, verb='unassigned', action_object=task,
+                        place_code=task.place.place_code)
+            messages.success(request, u"Task '%s' has been unassigned" % task.title)
+        else:
+            assignee = User.objects.get(id=kwargs['user_id'])
```

Comment:
See my comment on making the user to assign it to to be a form element. Then the form does this lookup and validation for you.